manager/manager.go

```diff
@@ -75,9 +92,27 @@ func (rm *realManager) Housekeep() {
 }
 
 func (rm *realManager) housekeep(start, end time.Time) {
-	data := rm.sources.ScrapeSources(start, end)
-	for _, p := range rm.processors {
-		data = p.Process(data)
+	if !start.Before(end) {
+		glog.Warningf("Wrong time provided to housekeep start:%s end: %s", start, end)
+		return
+	}
+
+	select {
+	case <-time.After(rm.resolution / 2):
+		glog.Warningf("Spent too long waiting for housekeeping to start")
+		return
+
+	case <-rm.housekeepSemaphoreChan:
+		// ok, good to go
 	}
-	rm.sink.ExportData(data)
+
+	go func(rm *realManager) {
+		// should always give back the semaphore
```

Comment:
We would have two functions that take start/end and do housekeeping. Lets leave it as is.